lib/tools/adb-commands.js

```diff
@@ -400,19 +404,23 @@ methods.toggleGPSLocationProvider = async function toggleGPSLocationProvider (en
  *        or to a restricted greylist for your target API level.
  */
 methods.setHiddenApiPolicy = async function setHiddenApiPolicy (value) {
-  await this.setSetting('global', 'hidden_api_policy_pre_p_apps', value);
-  await this.setSetting('global', 'hidden_api_policy_p_apps', value);
-  await this.setSetting('global', 'hidden_api_policy', value);
+  try {
+    await this.shell(HIDDEN_API_POLICY_KEYS.map((k) => `settings put global ${k} ${value}`).join(';'));
```

Comment:
it might be a good idea to have an option whether to throw an error if the method fails ot simply return true/false. This seems a bit wrong to me that we don't signal to the client in any way that the actual call has failed

Reply:
sounds good